skbio/sequence/_sequence.py

```diff
@@ -9,75 +9,66 @@
 from __future__ import absolute_import, division, print_function
 from future.builtins import range
 from future.utils import viewitems
-from six import string_types
+from future.standard_library import hooks
+from six import string_types, text_type
 
 import re
-from collections import Sequence, Counter, defaultdict
-from itertools import product
+import collections
+import numbers
+from contextlib import contextmanager
 
 import numpy as np
 from scipy.spatial.distance import hamming
 
 from skbio._base import SkbioObject
-from skbio.sequence import BiologicalSequenceError
+from skbio.util._misc import reprnator
 
+with hooks():
+    from itertools import zip_longest
 
-class BiologicalSequence(Sequence, SkbioObject):
-    """Base class for biological sequences.
+
+class Sequence(collections.Sequence, SkbioObject):
+    """Store biological sequence data and optional associated metadata.
+
+    ``Sequence`` objects do not enforce an alphabet and are thus the most
+    generic objects for storing biological sequence data. Subclasses ``DNA``,
+    ``RNA``, and ``Protein`` enforce the IUPAC character set [1]_ and provide
+    operations specific to each respective molecule type.
```

Comment:
*enforce the IUPAC character set [1]_ and provide operations specific to each respective molecule type.*

to 

*enforce the IUPAC character set [1]_ for, and provide operations specific to, each respective molecule type.*